pkg/healthcheck/healthcheck.go

```diff
@@ -898,10 +1009,79 @@ func (hc *HealthChecker) checkCustomResourceDefinitions() error {
 		objects = append(objects, &item)
 	}
 
-	return checkResources("CustomResourceDefinitions", objects, []string{"serviceprofiles.linkerd.io"})
+	return checkResources("CustomResourceDefinitions", objects, []string{"serviceprofiles.linkerd.io"}, shouldExist)
+}
+
+func (hc *HealthChecker) checkMutatingWebhookConfigurations(shouldExist bool) error {
+	options := metav1.ListOptions{
+		LabelSelector: k8s.ControllerNSLabel,
+	}
+	mwc, err := hc.kubeAPI.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().List(options)
+	if err != nil {
+		return err
+	}
+
+	objects := []runtime.Object{}
+	for _, item := range mwc.Items {
+		item := item // pin
+		objects = append(objects, &item)
+	}
+
+	return checkResources("MutatingWebhookConfigurations", objects, []string{k8s.ProxyInjectorWebhookConfigName}, shouldExist)
 }
 
-func checkResources(resourceName string, objects []runtime.Object, expectedNames []string) error {
+func (hc *HealthChecker) checkValidatingWebhookConfigurations(shouldExist bool) error {
+	options := metav1.ListOptions{
+		LabelSelector: k8s.ControllerNSLabel,
+	}
+	vwc, err := hc.kubeAPI.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().List(options)
+	if err != nil {
+		return err
+	}
+
+	objects := []runtime.Object{}
+	for _, item := range vwc.Items {
+		item := item // pin
+		objects = append(objects, &item)
+	}
+
+	return checkResources("ValidatingWebhookConfigurations", objects, []string{k8s.SPValidatorWebhookConfigName}, shouldExist)
+}
+
+func (hc *HealthChecker) checkPodSecurityPolicies(shouldExist bool) error {
+	options := metav1.ListOptions{
+		LabelSelector: k8s.ControllerNSLabel,
+	}
+	psp, err := hc.kubeAPI.PolicyV1beta1().PodSecurityPolicies().List(options)
+	if err != nil {
+		return err
+	}
+
+	objects := []runtime.Object{}
+	for _, item := range psp.Items {
+		item := item // pin
+		objects = append(objects, &item)
+	}
+
+	return checkResources("PodSecurityPolicies", objects, []string{fmt.Sprintf("linkerd-%s-control-plane", hc.ControlPlaneNamespace)}, shouldExist)
+}
+
+func checkResources(resourceName string, objects []runtime.Object, expectedNames []string, shouldExist bool) error {
+	if !shouldExist {
+		if len(objects) > 0 {
+			resources := ""
+			for _, obj := range objects {
+				m, err := meta.Accessor(obj)
+				if err != nil {
+					return err
+				}
+				resources += fmt.Sprintf("- %s/%s\n", resourceName, m.GetName())
```

Comment:
i'd avoid format-specific string rendering here, as this code is used by at least `linkerd check -o json`:

```
        {
          "description": "no ClusterRoleBindings exist",
          "hint": "https://linkerd.io/checks/#pre-l5d-existence",
          "error": "- ClusterRoleBindings/linkerd-linkerd-controller\n- ClusterRoleBindings/linkerd-linkerd-identity\n- ClusterRoleBindings/linkerd-linkerd-prometheus\n- ClusterRoleBindings/linkerd-linkerd-proxy-injector\n- ClusterRoleBindings/linkerd-linkerd-sp-validator\n- ClusterRoleBindings/linkerd-linkerd-tap\n",
          "result": "error"
        },
```

the newline-delimited list of resources looks really good when rendered as text, but i think it's more clear to just return an error string list of resources, which will render appropriately in all contexts. something like:
```
ClusterRoleBindings found but should not exist: linkerd-linkerd-controller linkerd-linkerd-identity linkerd-linkerd-prometheus linkerd-linkerd-proxy-injector linkerd-linkerd-sp-validator linkerd-linkerd-tap
```

if we wanted the best of both worlds, we probably should modify these APIs to return a structured object that enumerates the missing resource strings, but that's probably best done in a subsequent PR.